src/routes/formsg/formsgGGsRepair.ts

```diff
@@ -0,0 +1,264 @@
+    { data: { submissionId: string } },
+    never,
+    { submission: DecryptedContentAndAttachments }
+  > = async (req, res) => {
+    const { responses } = res.locals.submission.content
+
+    const requesterEmail = getField(responses, REQUESTER_EMAIL_FIELD)
+    const optionToSubmitCsv = getField(responses, OPTION_TO_SUBMIT_CSV)
+    const repoNames: string[] = []
+    if (optionToSubmitCsv === "Yes") {
+      const attachmentId = getId(responses, ATTACHMENT)
+      if (!attachmentId) {
+        throw new Error("No attachment id")
+      }
+      const decryptedFile: DecryptedFile =
+        res.locals.submission.attachments?.[attachmentId]
+      const reposCsv = Buffer.from(decryptedFile.content).toString()
+      if (!reposCsv.startsWith("repo_name")) {
+        logger.error("Invalid csv format")
+        return
+      }
+      const repos = reposCsv.split("\n").slice(1)
+      repos.forEach((repo) => {
+        repoNames.push(repo)
+      })
+      if (repoNames.length === 0) {
+        logger.error("No repo name provided")
+        return
+      }
+    } else {
+      const repoNamesFromTable = getFieldsFromTable(responses, REPO_NAME_FIELD)
+
+      if (!repoNamesFromTable) {
+        logger.error("No repo name provided")
+        return
+      }
+      repoNamesFromTable.forEach((repoName) => {
+        if (typeof repoName === "string") {
+          // actually wont happen based on our formsg form, but this code
+          // is added defensively
+          repoNames.push(repoName)
+        } else {
+          repoNames.push(repoName[0])
+        }
+      })
+    }
+
+    if (!requesterEmail?.endsWith("@open.gov.sg")) {
+      logger.error("Requester email is not from @open.gov.sg")
+      return
+    }
+    this.handleGGsFormSubmission(repoNames, requesterEmail)
+  }
+
+  handleGGsFormSubmission = (repoNames: string[], requesterEmail: string) => {
+    const repairs: ResultAsync<string, GitFileSystemError>[] = []
+
+    const clonedStagingRepos: string[] = []
+    const syncedStagingAndStagingLiteRepos: string[] = []
+    repoNames.forEach((repoName) => {
+      const repoUrl = `git@github.com:isomerpages/${repoName}`
+
+      repairs.push(
+        this.doesRepoNeedClone(repoName)
+          .andThen(() => {
+            const isStaging = true
+            return (
+              this.gitFileSystemService
+                // Repo does not exist in EFS, clone it
+                .cloneBranch(repoName, isStaging)
+                .andThen(() => {
+                  // take note of repos that cloned successfully
+                  clonedStagingRepos.push(repoName)
+                  return okAsync(true)
+                })
+            )
+          })
+          .orElse((error) => {
+            if (error === false) {
+              // Repo exists in EFS, no need to clone, but continue with syncing
+              return okAsync(false)
+            }
+            return errAsync<boolean, GitFileSystemError>(error)
```

Comment:
Nit: isn't this guaranteed to be a `GitFileSystemError`?

Reply:
76b0c5c93475d96d36384aa1aa6c402527ab51e9